src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -1721,1150 +2092,1070 @@ def _fail_if_component_with_name_type_already_shared(
+                hydrate=hydrate,
             )
 
-    def delete_flavor(self, flavor_id: UUID) -> None:
-        """Delete a flavor.
+    def update_authorized_device(
+        self, device_id: UUID, update: OAuthDeviceUpdate
+    ) -> OAuthDeviceResponse:
+        """Updates an existing OAuth 2.0 authorized device for internal use.
 
         Args:
-            flavor_id: The id of the flavor to delete.
+            device_id: The ID of the device to update.
+            update: The update to be applied to the device.
+
+        Returns:
+            The updated OAuth 2.0 authorized device.
 
         Raises:
-            KeyError: if the flavor doesn't exist.
-            IllegalOperationError: if the flavor is used by a stack component.
+            KeyError: If no device with the given ID exists.
         """
         with Session(self.engine) as session:
-            try:
-                flavor_in_db = session.exec(
-                    select(FlavorSchema).where(FlavorSchema.id == flavor_id)
-                ).one()
+            existing_device = session.exec(
+                select(OAuthDeviceSchema).where(
+                    OAuthDeviceSchema.id == device_id
+                )
+            ).first()
+            if existing_device is None:
+                raise KeyError(
+                    f"Unable to update device with ID {device_id}: No "
+                    "device with this ID found."
+                )
 
-                if flavor_in_db is None:
-                    raise KeyError(f"Flavor with ID {flavor_id} not found.")
-                components_of_flavor = session.exec(
-                    select(StackComponentSchema).where(
-                        StackComponentSchema.flavor == flavor_in_db.name
-                    )
-                ).all()
-                if len(components_of_flavor) > 0:
-                    raise IllegalOperationError(
-                        f"Stack Component `{flavor_in_db.name}` of type "
-                        f"`{flavor_in_db.type} cannot be "
-                        f"deleted as it is used by "
-                        f"{len(components_of_flavor)} "
-                        f"components. Before deleting this "
-                        f"flavor, make sure to delete all "
-                        f"associated components."
-                    )
-                else:
-                    session.delete(flavor_in_db)
-                    session.commit()
-            except NoResultFound as error:
-                raise KeyError from error
+            existing_device.update(update)
 
-    # -----
-    # Users
-    # -----
+            session.add(existing_device)
+            session.commit()
 
-    @classmethod
-    @lru_cache(maxsize=1)
-    def _get_resource_references(
-        cls,
-    ) -> List[Tuple[Type[SQLModel], str]]:
-        """Get a list of all other table columns that reference the user table.
+            return existing_device.to_model()
```

Comment:
As a rule of thumb I would expect `update` methods to return the same amount of information as `create` methods. So either you hydrate both or neither.

Reply:
This has changed now. All the update methods in the SQLZenStore return hydrated models now.